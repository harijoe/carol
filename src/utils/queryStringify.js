const queryStringify = (params) => {
  const keys = Object.keys(params)

  if (keys.length === 0) {
    return ''
  }

  return `?${keys.map(key => `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`).join('&')}`
}

export default queryStringify
