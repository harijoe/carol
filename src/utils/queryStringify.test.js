import queryStringify from './queryStringify'

describe('queryStringify', () => {

  it('returns the empty string when no params are passed in', () => {
    const queryString = queryStringify({})

    expect(queryString).toEqual('')
  })

  it('generates the correct query string', () => {
    const queryString = queryStringify({ username: "abcdef+1@gmail.com", password: 'qazio0!&' })

    expect(queryString).toMatchSnapshot()
  })

})
