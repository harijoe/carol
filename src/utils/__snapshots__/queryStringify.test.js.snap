exports[`queryStringify generates the correct query string 1`] = `"?username=abcdef%2B1%40gmail.com&password=qazio0!%26"`;
